### DISCUSSION TITLE
"Stop Training Text Encoder After" ?

### DISCUSSION THREAD

**Discussion by wktra** [d8ahazard/sd_dreambooth_extension#358] [Q&A]
What do I set this to?
Is this similar to the text encoder percentage in TheLastBen?

If so, do I set it to 100 for single character training?
@d8ahazard

**Comment by D4ndellion** [comment]
It has the same function as the text encoder percentage, but is not based on % but rather steps, so if you are training for 10 000 steps you need to set it to 10 000 for it to be 100%, 5000 for 50% and so on

**Reply by wktra** [reply]
> It has the same function as the text encoder percentage, but is not based on % but rather steps, so if you are training for 10 000 steps you need to set it to 10 000 for it to be 100%, 5000 for 50% and so on

brilliant! thank you~~~!!!

**Reply by wktra** [reply]
@dali99 sorry to bug you again, but I was getting great results before this added option was added to the extension. Now, it takes 9 times as long.  What setting do I leave _text encoder percentage_ at to get previous results in shorter times?

**Reply by nawnie** [reply]
try about 30% for style or pose and 50 for a person. if person is not able to be added to other styles use a lower percent, there is some play in what works which is why its a nice feature to have,

**Reply by wktra** [reply]
@nawnie thank you! and the setting for _prior loss weight_ should stay at the default 1, correct?

**Reply by nawnie** [reply]
Ya I wouldn't touch that, that's what's gonna keep all items of your class from looking like your subject ie all dogs look like snoopy.

Now if your doing something like waifu diffusion you may want to change that but... i.. wouldn't.

Another dream booth repo has had the text encoder customizable for awhile. You can train ut at 0 especially if you are captioning your own images. If not it will for you. It's main benefit is it's gonna use words it knows, sometimes if you Caption things your self you can throw a wrench in sd's vocabulary, if it's needed start by training ur model what the term is then load the images containing the term.. personally I call this pre training.

**Reply by wktra** [reply]
@dali99 i appreciate it. thank you.
unfortunately, since the last update, training is broken for me.

**Reply by D4ndellion** [reply]
see #366 for possible fixes

**Comment by d8ahazard** [comment]
So, for now, I've removed this feature, as it was breaking stuff. I'll maybe re-implement later, but if you want to "stop" training the text encoder after N steps, just train for that number of steps, cancel, and disable training the text encoder.

It's not super-elegant...but training actually works again.

f604af2

**Reply by D4ndellion** [reply]
I assume if you use something other than a constant learning rate that might have issues? Unless those values are based on the number of steps completed?

**Reply by wktra** [reply]
@dali99 I use polynomial with a warmup of 400. It provides the absolute best results for me. 
So as of right now, am I not supposed to use it? 
Is that why training has been broken?

**Reply by d8ahazard** [reply]
> @dali99 I use polynomial with a warmup of 400. It provides the absolute best results for me. So as of right now, am I not supposed to use it? Is that why training has been broken?

As of now, you can't use it, because it's been removed. ;)

And yes, that's what broke training.

**Reply by wktra** [reply]
@d8ahazard ahhh!
So then what's this setting do? 
<img width="412" alt="Screenshot 2022-11-29 183442" src="https://user-images.githubusercontent.com/117292904/204679045-b4f26069-1e8b-4a3d-8d63-2653ce4860a6.png">
"train text encoder"

should I deselect it if the training encoder steps is currently disabled?

**Reply by D4ndellion** [reply]
Only "Stop training text encoder after" has been removed, you can still train the text encoder

**Reply by Humanoidme** [reply]
Do you know if SD2.0 will still work after this fix? Thanks

**Reply by d8ahazard** [reply]
> Do you know if SD2.0 will still work after this fix? Thanks

No, 2.0 support is not fully integrated yet, at least for extracting checkpoints from the 768-v model. It *may* be possible to pull the 2.0 diffusers model from the hub, then train using a 768 resolution, but I haven't tried it yet. Purportedly, diffusers 0.9.0 works with it out of the box, which is what we're on now.

I also have this bit of code hidden in train_dreambooth.py that would probably need to be un-commented.

![image](https://user-images.githubusercontent.com/1633844/204831285-7dd6946b-890e-44f0-aaf9-fdd2650efde8.png)


And even if you can train it, I don't have support added for re-building them yet. I have local files that are a WIP, but nothing ready to push yet.
